### PR TITLE
Optimize fpUnaryMaskEvaluator with 3-operand instruction form

### PR DESCRIPTION
**Pull request by BradleyWood**
Signed-off-by: BradleyWood <bradley.wood@ibm.com>

### CODE REVIEW

**Comment by BradleyWood** [comment]
Jenkins build xlinux,win,osx,x32linux

**Comment by BradleyWood** [comment]
FYI, @0xdaryl

File: compiler/x/codegen/FPTreeEvaluator.cpp [hunk 1]
```diff
@@ -524,12 +524,18 @@ TR::Register *OMR::X86::TreeEvaluator::fpUnaryMaskEvaluator(TR::Node *node, TR::
       result->setIsSinglePrecision();
       }
 
-   // TODO 3-OP Optimization
-   if (result != value)
+   if (cg->comp()->target().cpu.supportsAVX())
       {
-      generateRegRegInstruction(TR::InstOpCode::MOVDQURegReg, node, result, value, cg);
+      generateRegRegMemInstruction(opcode, node, result, value, generateX86MemoryReference(cg->findOrCreate16ByteConstant(node, mask), cg), cg);
```
Comment (0xdaryl) [inline compiler/x/codegen/FPTreeEvaluator.cpp:529]: To save some shared library code space, can you hoist the `generateX86MemoryReference(cg->findOrCreate16ByteConstant(node, mask), cg)`  above the `if` and share it on both branches?

Comment (0xdaryl) [inline compiler/x/codegen/FPTreeEvaluator.cpp:529]: To share some .so code space, can you hoist the `generateX86MemoryReference(cg->findOrCreate16ByteConstant(node, mask), cg)` above the `if` and share it on both branches?

Comment (0xdaryl) [inline compiler/x/codegen/FPTreeEvaluator.cpp:529]: To share some .so code space, can you hoist the `generateX86MemoryReference(cg->findOrCreate16ByteConstant(node, mask), cg)` above the `if` and share it on both branches?

Comment (BradleyWood) [reply compiler/x/codegen/FPTreeEvaluator.cpp]: See [force-push](https://github.com/eclipse/omr/compare/9b2b12505e8257c586dbc2f6880f059f9ad2c364..bc9c56a47997673716c3a4e814c2bec93d61b2b0)

**Comment by BradleyWood** [comment]
Jenkins build xlinux,win,osx,x32linux

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**